caikit/core/model_manager.py

```diff
@@ -67,12 +69,119 @@ class ModelManager:
     def __init__(self):
         """Initialize ModelManager."""
         # Map to store module caches, to be used for singleton model lookups
-        self.singleton_module_cache = {}
-        self._singleton_lock = Lock()
+        self._singleton_module_cache = {}
+        self._trainers = {}
         self._finders = {}
         self._initializers = {}
+        self.__singleton_lock = Lock()
```

Comment:
This gets a double `__` so that the wrapper `contextmanager` can be `self._singleton_lock`